### DISCUSSION TITLE
I am trying to send image but i got an error

### DISCUSSION THREAD

**Discussion by sajidmirza0786** [netflie/whatsapp-cloud-api#16] [General]
Symfony\Component\HttpFoundation\Response::setContent(): Argument #1 ($content) must be of type ?string, Netflie\WhatsAppCloudApi\WhatsAppCloudApi given, called in /Applications/XAMPP/xamppfiles/htdocs/rn/ecommerce/vendor/laravel/framework/src/Illuminate/Http/Response.php on line 72

**Comment by sajidmirza0786** [comment]
<img width="1449" alt="Screenshot 2022-07-12 at 5 30 34 PM" src="https://user-images.githubusercontent.com/92370971/178485535-0afbd4c8-7192-4ac6-906a-83acf84d1131.png">
this is error screenshot

**Comment by boynet** [comment]
the easiest way is to upload the image to your server(by how ever you want) and send the image url
```
$link_id = new LinkID('http://www.example.com/image.jpg');
$whatsapp_cloud_api->sendImage('<destination-phone-number>', $link_id);
```

**Reply by sajidmirza0786** [reply]
$link_id = new LinkID('http://www.example.com/image.jpg');
$whatsapp_cloud_api->sendImage('<destination-phone-number>', $link_id);

**Reply by sajidmirza0786** [reply]
<img width="1920" alt="Screenshot 2022-07-14 121926" src="https://user-images.githubusercontent.com/92370971/178919361-54a0a2e5-da49-41c7-a2c1-fdc9edfdc987.png">
i have used status code is 200 but not getting message on whatsapp???

**Reply by boynet** [reply]
when you send text message it works?

**Reply by sajidmirza0786** [reply]
yes, now it is working fine all thank you

**Reply by man8414** [reply]
> when you send text message it works?

please could you help me
when i try send text message it didn't give error but the message doesn't send and the other person didn't receive it

**Reply by aalbarca** [reply]
> > when you send text message it works?
> 
> please could you help me when i try send text message it didn't give error but the message doesn't send and the other person didn't receive it

WhatsApp doesn't allow send messages if you didn't open before a window-session with some user. To open a conversation you need first to send a **template** message. This is a requirement from WhatsApp Cloud API.

**Comment by sajidmirza0786** [comment]
Yes it is working well but if you want to send attachment dynamically then
it is have some issue. If you want attachments or image you can add in
template statically
